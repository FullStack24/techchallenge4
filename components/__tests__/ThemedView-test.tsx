import * as React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';
import { ThemedView } from '../ThemedView';
import { useThemeColor } from '@/hooks/useThemeColor';

jest.mock('@/hooks/useThemeColor', () => ({
  useThemeColor: jest.fn(),
}));

it('deve renderizar ThemedView com a cor de fundo correta', () => {
  const mockThemeColor = '#FFFFFF';
  
  (useThemeColor as jest.Mock).mockReturnValue(mockThemeColor);

  const tree = renderer.create(
    <ThemedView lightColor="#FFFFFF" darkColor="#000000" style={{ padding: 10 }} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
