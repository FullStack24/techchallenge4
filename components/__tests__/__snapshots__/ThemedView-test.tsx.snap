// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar ThemedView com a cor de fundo correta 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "padding": 10,
      },
    ]
  }
/>
`;
